evm/multisig.js

```diff
@@ -0,0 +1,325 @@
+'use strict';
+
+require('dotenv').config();
+
+const {
+    Wallet,
+    getDefaultProvider,
+    utils: { isAddress, keccak256, Interface, formatEther },
+    constants: { AddressZero },
+    Contract,
+    BigNumber,
+} = require('ethers');
+const readlineSync = require('readline-sync');
+const { Command, Option } = require('commander');
+const {
+    printInfo,
+    printWalletInfo,
+    loadConfig,
+    isNumber,
+    isValidCalldata,
+    printWarn,
+    printError,
+    isStringArray,
+    isNumberArray,
+} = require('./utils');
+const IMultisig = require('@axelar-network/axelar-gmp-sdk-solidity/interfaces/IMultisig.json');
+const IGateway = require('@axelar-network/axelar-gmp-sdk-solidity/interfaces/IAxelarGateway.json');
+const IGovernance = require('@axelar-network/axelar-gmp-sdk-solidity/interfaces/IAxelarServiceGovernance.json');
+
+async function preExecutionChecks(multisigContract, multisigAction, wallet, target, calldata, nativeValue, yes) {
+    const isSigner = await multisigContract.isSigner(wallet.address);
+
+    if (!isSigner) {
+        throw new Error(`Caller ${wallet.address} is not an authorized multisig signer.`);
+    }
+
+    const multisigInterface = new Interface(multisigContract.interface.fragments);
+
+    let topic;
+
+    if (multisigAction === 'withdraw') {
+        topic = multisigInterface.encodeFunctionData('withdraw', [target, nativeValue]);
```

Comment:
```suggestion
        topic = multisigContract.interface.encodeFunctionData('withdraw', [target, nativeValue]);
```
could also do this to avoid creating a separate interface object